app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -470,10 +468,14 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun observeNewAttachments() = with(binding) {
-        newMessageViewModel.importedAttachments.observe(viewLifecycleOwner) { (attachments, importationResult) ->
-            attachmentAdapter.addAll(attachments)
-            attachmentsRecyclerView.isGone = attachmentAdapter.itemCount == 0
+    private fun observeNewAttachments() = with(newMessageViewModel) {
+        importedAttachments.observe(viewLifecycleOwner) { (newAttachments, importationResult) ->
+            attachmentAdapter.addAll(newAttachments)
+            newAttachments.filter { newAttachment ->
+                draft.attachments.none { it.uploadLocalUri == newAttachment.uploadLocalUri }
```

Comment:
Maybe we can do the filter in the `viewModel`. We're using only using `draft`, which is in the `viewModel` to filter anyway. 

That way, the view only need to display the value and all the processing is in the `viewModel`.

Also, it seems that `ImportationResult.SUCCESS` is never used.

Reply:
The check of Success value is unnecessary, because in both result value, we want to add the attachment anyway

Reply:
Yes. It's weird to have a status we return and we never use it anywhere.